### PR TITLE
Refactor types in Angular Demos p.4

### PR DESCRIPTION
**Pull request by GoodDayForSurf**

### CODE REVIEW

File: JSDemos/Demos/DataGrid/CascadingLookups/Angular/app/app.component.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { NgModule, Component, enableProdMode } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/CascadingLookups/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/CellEditingAndEditingAPI/Angular/app/app.component.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import {
   DxButtonModule,
 } from 'devextreme-angular';
 import ArrayStore from 'devextreme/data/array_store';
+import { DxDataGridTypes } from 'devextreme-angular/ui/data-grid';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/CellEditingAndEditingAPI/Angular/app/app.component.ts]: Group module, component and types to ui

File: JSDemos/Demos/DataGrid/CommandColumnCustomization/Angular/app/app.component.ts [hunk 3]
```diff
@@ -2,12 +2,15 @@ import { NgModule, Component, enableProdMode } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/CommandColumnCustomization/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/CustomNewRecordPosition/Angular/app/app.component.ts [hunk 4]
```diff
@@ -1,58 +1,61 @@
 import {
-  NgModule, Component, enableProdMode, ChangeDetectorRef,
+  NgModule, Component, enableProdMode,
 } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule, DxSelectBoxModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/CustomNewRecordPosition/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/CustomSummaries/Angular/app/app.component.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import {
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/CustomSummaries/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/CustomizeKeyboardNavigation/Angular/app/app.component.ts [hunk 6]
```diff
@@ -4,6 +4,7 @@ import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import {
   DxDataGridModule, DxButtonModule, DxCheckBoxModule, DxSelectBoxModule,
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/CustomizeKeyboardNavigation/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/DnDBetweenGrids/Angular/app/app.component.ts [hunk 7]
```diff
@@ -3,6 +3,8 @@ import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-bro
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/DnDBetweenGrids/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/EditStateManagement/Angular/app/app.component.ts [hunk 8]
```diff
@@ -7,6 +7,7 @@ import { HttpClientModule } from '@angular/common/http';
 
 import { DxDataGridModule, DxLoadPanelModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/EditStateManagement/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/FilterPanel/Angular/app/app.component.ts [hunk 9]
```diff
@@ -2,16 +2,18 @@ import { NgModule, Component, enableProdMode } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/FilterPanel/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/Filtering/Angular/app/app.component.ts [hunk 10]
```diff
@@ -29,61 +29,55 @@ export class AppComponent {
...
+
+  applyFilterTypes = [{
+    key: 'auto',
+    name: 'Immediately',
+  }, {
+    key: 'onClick',
+    name: 'On Button Click',
+  }];
+
+  saleAmountHeaderFilter: DxoHeaderFilterComponent['dataSource'] = [{
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/Filtering/Angular/app/app.component.ts:44]: @dxvladislavvolkov what do you think? may be not to use nested types?

File: JSDemos/Demos/DataGrid/FocusedRow/Angular/app/app.component.ts [hunk 11]
```diff
@@ -12,8 +12,9 @@ import {
   DxNumberBoxModule,
   DxCheckBoxModule,
 } from 'devextreme-angular';
-
+import { Options as DataSourceOptions } from 'devextreme/data/data_source';
 import 'devextreme/data/odata/store';
+import { DxDataGridTypes } from 'devextreme-angular/ui/data-grid';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/FocusedRow/Angular/app/app.component.ts]: Group module, component and types to ui

File: JSDemos/Demos/DataGrid/HorizontalVirtualScrolling/Angular/app/app.component.ts [hunk 12]
```diff
@@ -2,6 +2,8 @@ import { NgModule, Component, enableProdMode } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
+import { DxDataGridTypes } from 'devextreme-angular/ui/data-grid';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/HorizontalVirtualScrolling/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/InfiniteScrolling/Angular/app/app.component.ts [hunk 13]
```diff
@@ -1,8 +1,9 @@
 import { NgModule, Component, enableProdMode } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
-import { DxDataGridModule } from 'devextreme-angular';
-import { Service, Employee } from './app.service';
+import { DxDataGridComponent, DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/InfiniteScrolling/Angular/app/app.component.ts:4]: ui, module first

File: JSDemos/Demos/DataGrid/LocalReordering/Angular/app/app.component.ts [hunk 14]
```diff
@@ -1,11 +1,10 @@
 import {
-  NgModule, Component, enableProdMode, ViewChild,
+  NgModule, Component, enableProdMode,
 } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
-
 import { DxDataGridModule, DxCheckBoxModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/LocalReordering/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/MasterDetailAPI/Angular/app/app.component.ts [hunk 15]
```diff
@@ -2,6 +2,7 @@ import { NgModule, Component, enableProdMode } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule, DxTemplateModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/MasterDetailAPI/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/MultipleSorting/Angular/app/app.component.ts [hunk 16]
```diff
@@ -4,8 +4,8 @@ import {
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridComponent, DxDataGridModule, DxCheckBoxModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/MultipleSorting/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/Overview/Angular/app/app.component.ts [hunk 17]
```diff
@@ -9,6 +9,7 @@ import {
   DxTemplateModule,
 } from 'devextreme-angular';
 import DataSource from 'devextreme/data/data_source';
+import { DxDataGridTypes } from 'devextreme-angular/ui/data-grid';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/Overview/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/PDFCellCustomization/Angular/app/app.component.ts [hunk 18]
```diff
@@ -6,6 +6,7 @@ import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/PDFCellCustomization/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/PDFExportImages/Angular/app/app.component.ts [hunk 19]
```diff
@@ -6,6 +6,7 @@ import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/PDFExportImages/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/PDFHeaderAndFooter/Angular/app/app.component.ts [hunk 20]
```diff
@@ -3,12 +3,12 @@ import {
 } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
-
 import {
   DxDataGridModule, DxTemplateModule,
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/PDFHeaderAndFooter/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/PDFOverview/Angular/app/app.component.ts [hunk 21]
```diff
@@ -6,6 +6,7 @@ import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/PDFOverview/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/RecalculateWhileEditing/Angular/app/app.component.ts [hunk 22]
```diff
@@ -2,6 +2,7 @@ import { NgModule, Component, enableProdMode } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/RecalculateWhileEditing/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/RemoteReordering/Angular/app/app.component.ts [hunk 23]
```diff
@@ -4,6 +4,7 @@ import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/RemoteReordering/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/ToolbarCustomization/Angular/app/app.component.ts [hunk 24]
```diff
@@ -11,6 +11,7 @@ import {
   DxButtonModule,
 } from 'devextreme-angular';
 import query from 'devextreme/data/query';
+import { DxSelectBoxTypes } from 'devextreme-angular/ui/select-box';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/ToolbarCustomization/Angular/app/app.component.ts]: Group module and types to ui

File: JSDemos/Demos/DataGrid/VirtualScrolling/Angular/app/app.component.ts [hunk 25]
```diff
@@ -2,7 +2,9 @@ import { NgModule, Component, enableProdMode } from '@angular/core';
 import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { DxDataGridModule } from 'devextreme-angular';
```
Comment (mpreyskurantov) [inline JSDemos/Demos/DataGrid/VirtualScrolling/Angular/app/app.component.ts]: Group module and types to ui

**Review by mpreyskurantov — CHANGES_REQUESTED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Review by mpreyskurantov — APPROVED** [review]

**Pull request merged**